### PR TITLE
Fix default value handling for false and 0 values

### PR DESCRIPTION
**Pull request by andresmeidla**
`if (joi._flags && joi._flags.default) ` 
would not let through false (boolean) and 0 (number) values,
added explicit check for undefined and null to fix it

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,14 +4,15 @@
...
+    "postinstall": "npm run build",
     "build": "6to5 -s -e src --out-dir build",
     "pretest": "npm run build",
     "test": "mocha",
     "prepublish": "npm run build"
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/lightsofapollo/joi-to-json-schema.git"
+    "url": "git+https://github.com/andresmeidla/joi-to-json-schema.git"
```
Comment (lightsofapollo) [inline package.json]: ?

File: package.json [hunk 2]
```diff
@@ -25,14 +26,17 @@
   ],
   "license": "Apache2",
   "bugs": {
-    "url": "https://github.com/lightsofapollo/joi-to-json-schema/issues"
+    "url": "https://github.com/andresmeidla/joi-to-json-schema/issues"
   },
-  "homepage": "https://github.com/lightsofapollo/joi-to-json-schema",
+  "homepage": "https://github.com/andresmeidla/joi-to-json-schema",
```
Comment (lightsofapollo) [inline package.json]: Looks like these (and below) changes are unrelated can you remove them ?

**Comment by lightsofapollo** [comment]
looks good but please remove the package.json changes.

**Comment by andresmeidla** [comment]
removed the package.json changes

**Comment by lightsofapollo** [comment]
released in 2.2.1
